src/package-operations.ts

```diff
@@ -159,10 +163,51 @@ export async function updatePackage(
   packageMetadata: { dirPath: string; manifest: Partial<PackageManifest> },
   updateSpecification: UpdateSpecification | MonorepoUpdateSpecification,
 ): Promise<void> {
-  await writeJsonFile(
-    pathUtils.join(packageMetadata.dirPath, PACKAGE_JSON),
-    getUpdatedManifest(packageMetadata.manifest, updateSpecification),
-  );
+  await Promise.all([
+    writeJsonFile(
+      pathUtils.join(packageMetadata.dirPath, PACKAGE_JSON),
+      getUpdatedManifest(packageMetadata.manifest, updateSpecification),
+    ),
+    updateSpecification.shouldUpdateChangelog
+      ? updatePackageChangelog(packageMetadata, updateSpecification)
+      : Promise.resolve(),
+  ]);
+}
+
+/**
+ * Updates the changelog file of the given package, using
+ * @metamask/auto-changelog. Assumes that the changelog file is located at the
+ * package root directory and named "CHANGELOG.md".
+ *
+ * @param packageMetadata - The metadata of the package to update.
+ * @param updateSpecification - The update specification, which determines how
+ * the update is performed.
+ */
+async function updatePackageChangelog(
+  packageMetadata: { dirPath: string },
+  updateSpecification: UpdateSpecification | MonorepoUpdateSpecification,
+) {
+  const { dirPath: projectRootDirectory } = packageMetadata;
+  const { newVersion, repositoryUrl } = updateSpecification;
+
+  let changelogContent: string;
+  const changelogPath = `${projectRootDirectory}/CHANGELOG.md`;
```

Comment:
Nit: Probably better to use `path.join` here for Windows compatibility.

Reply:
Done in 19c3a07